regolith/helpers/makeappointmentshelper.py

```diff
@@ -78,16 +86,54 @@ def construct_global_ctx(self):
     def sout(self):
         rc = self.rc
         gaps = []
+        outdated = []
+        depleted = []
+
+        if rc.begin_date and rc.date:
+            raise ValueError("Please enter either an interval or a date, not both")
+        if rc.date:
+            desired_date = date_parser.parse(rc.date).date()
+        else:
+            desired_date = dt.date.today()
 
         for person in self.gtx["people"]:
-            appointments = person.get("appointments")
-            total_load = 0
-            for appt in appointments:
-                if rc.gap and is_current(appt):
-                    total_load += appt.get("loading")
-            if rc.gap and total_load != 1:
-                gaps.append(person)
-        for person in gaps:
-            print(person.get('name'))
+            appts = person.get("appointments")
+            if rc.gap:
+                if rc.begin_date:
+                    if not is_fully_appointed(appts, begin=rc.begin_date, duration=rc.duration)[0]:
+                        gaps.append(person.get('name'))
+                        gaps.append((is_fully_appointed(appts, begin=rc.begin_date, duration=rc.duration))[1])
+                else:
+                    if not is_fully_appointed(appts, now=desired_date)[0]:
+                        gaps.append(person.get('name'))
+                        gaps.append((is_fully_appointed(appts, now=desired_date))[1])
+            if rc.out_of_date:
+                for appt in appts:
+                    grant = fuzzy_retrieval(self.gtx["grants"], ['_id'], appt.get("grant"), case_sensitive=False)
```

Comment:
I think if you know the ID you can use rc.client.get_one() and it will be quicker I think

Reply:
I can't seem to find this function used elsewhere in the regolith project, but deducing the pattern from `rc.client.add_one()`, it looks like to retrieve the grant here it should be `rc.client.get_one(rc.database, "grants", {"_id": appt.get("grant")})`? Also, I do see another function `rc.client.find_one()` - is that maybe the one useful here?

Reply:
yes.  I haven't used these functions much, but I think they may be useful.  Could you look  into it?  fuzzy_retrieval is very slow because it iterates through the whole collection looking for something.  If we know for sure the ID we should be able to just fetch the document with that ID and I think that is what those rc.client functions do .  they are probably in runcontrol.py

Reply:
thank you! will do